### PR TITLE
[CFD-39] Push notifications

### PR DESCRIPTION
**Pull request by jasonfyw**
### Description

Backend: creates endpoint to send a notification to a given list of devices

Frontend: adds methods to register the device for notifications, as well as an example component that can display incoming notifications and send a notification to self.

### Reason for Change

Completes CFD-39

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Testing

Using the push token displayed on the Expo app, notifications can successfully be sent when tested with the following methods:
- [Expo's push notification tool](https://expo.dev/notifications)
- tRPC panel
- The send notification button on the Expo app

### CODE REVIEW

**Comment by BakerWJ** [comment]
hmmmm not sure why actions aren't being activated -- maybe try making an empty commit + pushing

File: apps/expo/package.json
Comment (BakerWJ) [inline apps/expo/package.json:56]: why did this change happen?

File: packages/api/src/router/notification.ts
Comment (BakerWJ) [inline packages/api/src/router/notification.ts:37]: can use a `map` here

File: packages/api/src/router/notification.ts
Comment (BakerWJ) [inline packages/api/src/router/notification.ts:19]: this might be nice to keep for testing notifications in the future 🤔 -- I'm assuming that the blocker before integrating this with something like the newPost endpoint is that we don't have the user pushTokens stored in the db right now?

**Review by BakerWJ — COMMENTED** [review]
will look in a bit more depth later -- we have a draft pr for implementing auth right now, so we might wait to merge this in until we have a user system set up (if that is one of the blockers for push tokens)

File: .vscode/settings.json
Comment (BakerWJ) [inline .vscode/settings.json:19]: I think there is a way to make this your personal setting instead of editing the project `settings.json`

File: apps/expo/app.config.ts
Comment (BakerWJ) [inline apps/expo/app.config.ts:34]: maybe this should be an environment variable? unfortunately we can't all join the same project for expo :(

File: apps/expo/src/app/notifications.tsx
Comment (BakerWJ) [inline apps/expo/src/app/notifications.tsx:1]: While good for testing, I don't think that we need this page on the mobile app (since most or all of our notifications are being sent from web)

File: apps/expo/src/notifications/registerNotifications.tsx
Comment (BakerWJ) [inline apps/expo/src/notifications/registerNotifications.tsx:16]: is there another check required for iOS, or is this only needed for android?

File: apps/expo/src/notifications/registerNotifications.tsx
Comment (BakerWJ) [inline apps/expo/src/notifications/registerNotifications.tsx:31]: maybe add this check to the start of the function (also probably want to throw an error)

File: apps/expo/src/notifications/registerNotifications.tsx
Comment (BakerWJ) [inline apps/expo/src/notifications/registerNotifications.tsx:28]: you can just do a return here instead of having the `let token` at the top

File: packages/api/src/router/notification.ts
Comment (BakerWJ) [inline packages/api/src/router/notification.ts:10]: this should probably be an `adminProcedure` if we keep it

**Review by BakerWJ — COMMENTED** [review]
looks good overall -- just some minor changes + questions

File: .vscode/settings.json
Comment (jasonfyw) [reply .vscode/settings.json:19]: I use `tabSize` 4 and `detectIndentation` false for most other projects. Is there a way to modify workspace settings without changing `settings.json`?

File: apps/expo/src/app/notifications.tsx
Comment (jasonfyw) [reply apps/expo/src/app/notifications.tsx:1]: Ok, should I scrap this component then?

File: apps/expo/src/notifications/registerNotifications.tsx
Comment (jasonfyw) [reply apps/expo/src/notifications/registerNotifications.tsx:16]: The Expo docs only showed a check for android. It works fine on iOS so I assume it's fine

File: .vscode/settings.json
Comment (BakerWJ) [reply .vscode/settings.json:19]: can this go in your user settings instead? the only rules that should be in the project `settings.json`should be project-specific (ie linting + workspace configuration)

File: apps/expo/src/app/notifications.tsx
Comment (BakerWJ) [reply apps/expo/src/app/notifications.tsx:1]: If the expo push notification tool has the same functionality, then I think we can delete it

File: .vscode/settings.json
Comment (jasonfyw) [reply .vscode/settings.json:19]: Fair enough. I'll take it out and just change the settings on my end

File: apps/expo/src/notifications/registerNotifications.tsx
Comment (BakerWJ) [inline apps/expo/src/notifications/registerNotifications.tsx:1]: this should be a ts file

File: packages/db/package.json
Comment (BakerWJ) [inline packages/db/package.json:21]: can u change this version back

**Review by BakerWJ — COMMENTED** [review]
good to merge after updating these 2 comments -- can you also add something to .env.example for the expo project id?

File: packages/api/src/router/notification.ts
Comment (BakerWJ) [inline packages/api/src/router/notification.ts:39]: lol not gonna block on this but in next pr use the logger instead of console

File: apps/expo/src/notifications/registerNotifications.ts
Comment (BakerWJ) [inline apps/expo/src/notifications/registerNotifications.ts:31]: i wonder if push notifications even work on simulator?

**Review by BakerWJ — DISMISSED** [review]

File: packages/api/src/router/notification.ts
Comment (BakerWJ) [inline packages/api/src/router/notification.ts:41]: `logger.error` 🥲

**Review by BakerWJ — APPROVED** [review]
would recommend turning on auto merge bc of ios build

File: packages/api/src/router/notification.ts
Comment (jasonfyw) [reply packages/api/src/router/notification.ts:41]: Oh my bad lol I'll use the logger in my next pr

**Pull request merged**
